Build failure: devenv

### Nixpkgs version

- Stable (25.05)

### Steps to reproduce

Include pkgs.devenv into environment.systemPackages and execute `nixos-rebuild switch`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/312729093

### Relevant log output

```console
devenv>    Compiling devenv-cache-core v1.9.1 (/build/source/devenv-cache-core)
devenv>    Compiling devenv-eval-cache v1.9.1 (/build/source/devenv-eval-cache)
devenv>    Compiling devenv-tasks v1.9.1 (/build/source/devenv-tasks)
devenv> error[E0658]: `let` expressions in this position are unstable
devenv>   --> devenv-eval-cache/src/command.rs:97:16
devenv>    |
devenv> 97 |               && let Ok(Some(output)) =
devenv>    |  ________________^
devenv> 98 | |                 query_cached_output(self.pool, &cmd_hash, &self.extra_paths).await
devenv>    | |__________________________________________________________________________________^
devenv>    |
devenv>    = note: see issue #53667 <https://github.com/rust-lang/rust/issues/53667> for more information
devenv>
devenv> error[E0658]: `let` expressions in this position are unstable
devenv>    --> devenv-eval-cache/src/internal_log.rs:170:12
devenv>     |
devenv> 170 |           if let InternalLog::Msg {
devenv>     |  ____________^
devenv> 171 | |             level: Verbosity::Error,
devenv> 172 | |             msg,
devenv> 173 | |             ..
devenv> 174 | |         } = self
devenv>     | |________________^
devenv>     |
devenv>     = note: see issue #53667 <https://github.com/rust-lang/rust/issues/53667> for more information
devenv>
devenv> error[E0658]: `let` expressions in this position are unstable
devenv>    --> devenv-eval-cache/src/internal_log.rs:185:12
devenv>     |
devenv> 185 |           if let InternalLog::Msg {
devenv>     |  ____________^
devenv> 186 | |             level: Verbosity::Error,
devenv> 187 | |             msg,
devenv> 188 | |             ..
devenv> 189 | |         } = self
devenv>     | |________________^
devenv>     |
devenv>     = note: see issue #53667 <https://github.com/rust-lang/rust/issues/53667> for more information
devenv>
devenv> error[E0658]: `let` expressions in this position are unstable
devenv>    --> devenv-tasks/src/task_state.rs:122:16
devenv>     |
devenv> 122 |             if let Some(env) = value.get("devenv").and_then(|d| d.get("env"))
devenv>     |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
devenv>     |
devenv>     = note: see issue #53667 <https://github.com/rust-lang/rust/issues/53667> for more information
devenv>
devenv> error[E0658]: `let` expressions in this position are unstable
devenv>    --> devenv-tasks/src/task_state.rs:123:20
devenv>     |
devenv> 123 |                 && let Some(env_obj) = env.as_object()
devenv>     |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
devenv>     |
devenv>     = note: see issue #53667 <https://github.com/rust-lang/rust/issues/53667> for more information
devenv>
devenv> error[E0658]: `let` expressions in this position are unstable
devenv>    --> devenv-tasks/src/tasks.rs:441:40
devenv>     |
devenv> 441 | ...                   && let Some(output_value) = output.as_object()
devenv>     |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
devenv>     |
devenv>     = note: see issue #53667 <https://github.com/rust-lang/rust/issues/53667> for more information
devenv>
devenv> For more information about this error, try `rustc --explain E0658`.
devenv> error: could not compile `devenv-eval-cache` (lib) due to 3 previous errors
devenv> warning: build failed, waiting for other jobs to finish...
devenv> error: could not compile `devenv-tasks` (lib) due to 3 previous errors
error: builder for '/nix/store/h6nl8bx7kig40s7gz81yvcn7g8x6ifar-devenv-1.9.1.drv' failed with exit code 101
error: 1 dependencies of derivation '/nix/store/q5jj0haksjnaa7qjl49rm9b3gbyjq603-devenv-1.9.1_fish-completions.drv' failed to build
error: 1 dependencies of derivation '/nix/store/32w5d4i7l0hp5a32kp6z64mp1idzzfhl-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/swr5mxidp5m1r3ixgmq78fg1mrilj55v-nixos-system-livefish-laptop-25.05.20251107.6faeb06.drv' failed to build
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.17.7, NixOS, 25.05 (Warbler), 25.05.20251104.ca534a7`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.5`
 - channels(root): `"nixos-25.05"`
 - nixpkgs: `/nix/store/5xlv8zvz1m19x9hx1ilyw0as7z7l8q4x-source`

### Notify maintainers

@domenkozar 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/de/devenv/package.nix
+++ b/pkgs/by-name/de/devenv/package.nix
@@ -3,10 +3,11 @@
   fetchFromGitHub,
   makeBinaryWrapper,
   installShellFiles,
-  rustPlatform,
+  rustPackages_1_89,
   testers,
   cachix,
   nixVersions,
+  gitMinimal,
   openssl,
   dbus,
   pkg-config,
@@ -15,7 +16,7 @@
 }:
 
 let
-  version = "1.9.1";
+  version = "1.10";
   devenvNixVersion = "2.30.4";
 
   devenv_nix =
@@ -34,7 +35,7 @@ let
         __intentionallyOverridingVersion = true;
       });
 in
-rustPlatform.buildRustPackage {
+rustPackages_1_89.rustPlatform.buildRustPackage {
   pname = "devenv";
   inherit version;
 
@@ -42,10 +43,10 @@ rustPlatform.buildRustPackage {
     owner = "cachix";
     repo = "devenv";
     tag = "v${version}";
-    hash = "sha256-v86pQGIWHJPkRryglJSXOp0aEoU6ZtURuURsXLqfqSE=";
+    hash = "sha256-rsb+6Wca43guzLL4Czoc89L394ZW9JZF2MShxaz2Sx4=";
   };
 
-  cargoHash = "sha256-41VmzZvoRd2pL5/o6apHztpS2XrL4HGPIJPBkUbPL1I=";
+  cargoHash = "sha256-Wt47YdBEtFXQACk1ByDwQyXzHU4/nGVQKY7gaZeQrQ4=";
 
   buildAndTestSubdir = "devenv";
 
@@ -59,6 +60,16 @@ rustPlatform.buildRustPackage {
     openssl
     dbus
   ];
+
+  nativeCheckInputs = [
+    gitMinimal
+  ];
+
+  preCheck = ''
+    git init
+    git config user.email "test@example.com"
+    git config user.name "Test User"
+  '';
 
   postInstall =
     let

